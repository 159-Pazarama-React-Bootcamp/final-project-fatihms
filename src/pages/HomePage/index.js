import React from "react";

import { Link } from "react-router-dom";

import styles from "./styles.module.css";

function HomePage() {
  return (
    <div className={styles.homepage}>
      <main>
        <Link to="/basvuru-olustur" className={styles["homepage-link"]}>
          <button type="button">Başvuru Oluştur</button>
        </Link>
        <Link to="/basvuru-sorgula" className={styles["homepage-link"]}>
          <button type="button">Başvuru Sorgula</button>
        </Link>
      </main>
    </div>
  );
}

export default HomePage;
