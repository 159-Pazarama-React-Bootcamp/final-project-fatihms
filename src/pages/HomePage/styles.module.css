.homepage {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

.homepage > main {
  display: flex;
  flex-direction: column;
}

.homepage > main > .homepage-link > button {
  border-radius: 8px;
  border: 2px solid #ccc;
  margin-bottom: 4px;
  padding: 25px 50px;
  cursor: pointer;
  font-size: 18px;
}
