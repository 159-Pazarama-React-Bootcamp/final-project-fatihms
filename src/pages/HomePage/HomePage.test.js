/*eslint-disable */

import React from "react";

import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { MemoryRouter as Router } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom/extend-expect";

import HomePage from "./index";

describe("HomePage", () => {
  beforeEach(() => {
    render(
      <Router>
        <HomePage />
      </Router>
    );
  });
  test("HomePage renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <HomePage />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("Başvuru Oluştur button renders correctly", () => {
    expect(screen.getByText("Başvuru Oluştur")).toBeInTheDocument();
  });
  test("Başvuru Sorgula button renders correctly", () => {
    expect(screen.getByText("Başvuru Sorgula")).toBeInTheDocument();
  });
});
