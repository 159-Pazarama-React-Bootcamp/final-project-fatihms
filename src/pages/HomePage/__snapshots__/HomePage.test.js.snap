// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage HomePage renders correctly 1`] = `
<div
  className="homepage"
>
  <main>
    <a
      className="make-application-link"
      href="/basvuru-olustur"
      onClick={[Function]}
    >
      <button
        type="button"
      >
        Başvuru Oluştur
      </button>
    </a>
    <a
      className="application-status-link"
      href="/basvuru-sorgula"
      onClick={[Function]}
    >
      <button
        type="button"
      >
        Başvuru Sorgula
      </button>
    </a>
  </main>
</div>
`;
